subiquity/common/serialize.py

```diff
@@ -192,9 +192,10 @@ def _deserialize_attr(self, annotation, value, metadata, path):
             return annotation(*args)
         else:
             args = {}
-            for field in attr.fields(annotation):
+            fields = {field.name: field for field in attr.fields(annotation)}
+            for key in value.keys():
```

Comment:
I tend to write this sort of thing as "for k, v in value.items()" but well, super minor.